integrations/dell_idrac/dell_idrac_test.py

```diff
@@ -0,0 +1,203 @@
+
+    def test_change_password_missing_user_id(self):
+        plugin = self.plugin()
+
+        plugin.get_config = MagicMock(side_effect=lambda key, single=True: {
+            "login": "admin",
+            "password": "adminpass",
+            "idrac_ip": "1.2.3.4",
+            "user_id": None
+        }.get(key))
+
+        with patch.object(DelliDRACClient, "get_user_id_from_user_fields", return_value=None):
+            with self.assertRaises(SaasException) as ctx:
+                plugin.change_password()
+            self.assertIn("Cannot determine user_id", str(ctx.exception))
+
+    def test_change_password_username_mismatch(self):
+        plugin = self.plugin()
+        with patch.object(DelliDRACClient, "get_user_id_from_user_fields", return_value="42"), \
+             patch.object(DelliDRACClient, "check_username_by_id", side_effect=SaasException("Username mismatch with user_id")):
+            with self.assertRaises(SaasException) as ctx:
+                plugin.change_password()
+            self.assertIn("Username mismatch with user_id", str(ctx.exception))
+
+    def test_change_password_http_error(self):
+        plugin = self.plugin()
+        with patch.object(DelliDRACClient, "get_user_id_from_user_fields", return_value="42"), \
+             patch.object(DelliDRACClient, "check_username_by_id"), \
+             patch.object(DelliDRACClient, "change_dell_idrac_user_password", side_effect=SaasException("Password rotation failed: 500")):
+            with self.assertRaises(SaasException) as ctx:
+                plugin.change_password()
+            self.assertIn("Password rotation failed", str(ctx.exception))
+
+    def test_rollback_password_success(self):
+        plugin = self.plugin(prior_password=Secret("OldPassword123"))
+        plugin._client = MagicMock()
+        plugin._SaasPlugin__user_id = "42"
+        plugin.user.prior_password = Secret(("SomeOtherPassword", "OldPassword123"))
+        plugin._client.change_dell_idrac_user_password = MagicMock()
+        plugin.rollback_password()
+        plugin._client.change_dell_idrac_user_password.assert_called_once_with("42", "OldPassword123")
+
+    def test_rollback_password_no_prior(self):
+        plugin = self.plugin(prior_password=None)
+        plugin._client = MagicMock()
+        plugin._SaasPlugin__user_id = "42"
+        plugin.user.prior_password = None
+        with self.assertRaises(Exception):
+            plugin.rollback_password()
+
+    def test_rollback_password_http_error(self):
+        plugin = self.plugin(prior_password=Secret("OldPassword123"))
+        plugin._client = MagicMock()
+        plugin._SaasPlugin__user_id = "42"
+        plugin.user.prior_password = Secret(("SomeOtherPassword", "OldPassword123"))
+        plugin._client.change_dell_idrac_user_password.side_effect = SaasException("Password change request failed: 500")
+        with self.assertRaises(SaasException) as ctx:
```

Comment:
Wherever you are expecting the Exceptions, shouldn't you be validating the error messages.